python/cuml/internals/device_support.pyx

```diff
@@ -15,11 +15,24 @@
 #
 
 
+from packaging.version import Version
+
+MIN_SKLEARN_VERSION = Version(1.5)
+
+
 try:
     import sklearn  # noqa: F401  # no-cython-lint
+
     CPU_ENABLED = True
+
+    if(Version(sklearn.__version__) >= MIN_SKLEARN_VERSION):
+        MIN_SKLEARN_PRESENT = True
+    else:
+        MIN_SKLEARN_PRESENT = False
```

Comment:
Wondering if we shouldn't just set `CPU_ENABLED = False`. If the version is too low, we can't use it anyway, right?

Reply:
I debated about that for a while, but in general it seems that even say with Sciki 1.2, 95% of functionality would still work. It’d be untested, and probably wouldn’t affect a lot of users. So leaned more into being a bit more permissive, though a bit unsure if we should throw a warning here. 

That said, I could be convinced otherwise.